catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/MgrsCoordinateProcessor.java

```diff
@@ -0,0 +1,347 @@
+      String mgrsPart,
+      MgrsPartType previousType,
+      StringBuilder mgrsBuilder,
+      List<String> mgrsStrings) {
+    LOGGER.trace("Found a NUMERIC [{}] which occurs after a {}", mgrsPart, previousType);
+    switch (previousType) {
+      case ZONE:
+      case HUNDRED_KM:
+      case NUMERIC:
+        String processed = processNumericType(mgrsPart);
+        LOGGER.trace(
+            "NUMERIC [{}] processed = [{}], storing final result coordinate", mgrsPart, processed);
+        mgrsStrings.add(mgrsBuilder.toString() + processed);
+        break;
+      case NONE:
+        LOGGER.trace(
+            "Ignoring this part [{}], since NUMERIC is only valid in a ZONE but no ZONE was provided",
+            mgrsPart);
+    }
+  }
+
+  /**
+   * Given an MGRS {@link MgrsPartType#NUMERIC} part format the string to ensure an accurate
+   * conversion, using whitespace as a hint as to the user's intentions.
+   *
+   * <p>The usng4j library demands a minimum of 6 symbols if the easting / northing data is not to
+   * be ignored. That is, the easting component must be fully specified for the conversion to work.
+   * To accomodate this, zeros are prepended onto the easting to whatever appropriate degree.
+   *
+   * <p>For example, an input of <span>4Q KJ 1 2</span> will become <span>4QKJ000012</span> so the
+   * easting / northing is not ignored.
+   *
+   * <p>Refer to the unit tests for more comprehensive examples.
+   *
+   * <p>TODO DDF-4243 - This should be simplified if usng4j is improved
+   *
+   * @see #assemble(String, String)
+   * @param mgrsPart an MGRS substring that contains BOTH the easting and northing.
+   * @return a modified easting/northing string with zeros backfilled to ensure accurate conversion.
+   */
+  @SuppressWarnings("squid:S1135" /* Ticket number provided on to-do comment */)
+  private static String processNumericType(String mgrsPart) {
+    final String[] numericParts = mgrsPart.split("[ ]+");
+    if (numericParts.length >= 3 || numericParts.length == 0) {
+      throw new IllegalArgumentException(
+          "The mgrs part provided is not a valid NUMERIC type: " + mgrsPart);
+    }
+    if (numericParts.length == 2) {
+      return assemble(numericParts[0], numericParts[1]);
+    }
+    String eastingAndNorthing = numericParts[0];
+    if (eastingAndNorthing.length() % 2 > 0) {
+      eastingAndNorthing = eastingAndNorthing.substring(0, eastingAndNorthing.length() - 1);
+    }
+    final int halfway = eastingAndNorthing.length() / 2;
+    final String easting = eastingAndNorthing.substring(0, halfway);
+    final String northing = eastingAndNorthing.substring(halfway);
+    return assemble(easting, northing);
+  }
+
+  private static String assemble(String easting, String northing) {
+    return getZeroPaddedCoordinateString(easting) + getZeroPaddedCoordinateString(northing);
+  }
+
+  private static String getZeroPaddedCoordinateString(String input) {
+    if (input.length() < MAX_COORDINATE_LENGTH) {
```

Comment:
✏️ So you don't actually need to perform this test since...

`StringUtils.repeat("a", -2) = ""`

This method can be...
```
private static String getZeroPaddedCoordinateString(String input) {
    return StringUtils.repeat(ZERO, MAX_COORDINATE_LENGTH - input.length()) + input;
}
```

Reply:
or even better...

```
private static String getZeroPaddedCoordinateString(String input) {
    return StringUtils.leftPad(input, ZERO, MAX_COORDINATE_LENGTH - input.length());
}
```